### PR TITLE
Changes for HT dashboard

### PR DESCRIPTION
**Pull request by hmzahmed**
- Changed Colors for Quizzes
- Changed Colors for Lesson Plans
- Changed View For Diagnostic Test
- Changed View for Midpoint Test
- Changed View for Summative Test
- Added Tabs for Comparison Matrix
- Added the functionality which enables the behavior that only questions whose grade is equal or lower that that of the student grade are counted when calculating percentage

### CODE REVIEW

**Review by mudssrali — CHANGES_REQUESTED** [review]
Overall changes look fine to me. Thanks @hmzahmed - However, please make sure to read all comments and make a habit, to not do it again in future PRs. These are general things, we need to keep them on our radar always while working. To write good code:

- Follow the convention e.g. camelCasing
- Whenever possible and need, make reusable constructs
- Don't Repeat Yourself, create once and use it where you need
- Good variable names
- Align with latest changes in the technology.

File: web/src/modules/TIP/HTDashboard/Tests/MidpointTest/index.tsx [hunk 1]
```diff
@@ -14,15 +13,15 @@ enum ActiveCard {
...
 
 const ordered_groups: Array<OrderedGroupItem> = [
-	{ group: 'KG', color: 'Blue' },
-	{ group: '1', color: 'Yellow' },
-	{ group: '2', color: 'Green' },
-	{ group: '3', color: 'Orange' }
+	{ level: 'Level KG', color: 'Blue' },
+	{ level: 'Level 1', color: 'Yellow' },
+	{ level: 'Level 2', color: 'Green' },
+	{ level: 'Level 3', color: 'Orange' }
```
[the remark this answers is not in the document]
Comment (hmzahmed) [reply web/src/modules/TIP/HTDashboard/Tests/MidpointTest/index.tsx:24]: Yes

File: web/src/hooks/useTIPStudents.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React, { useMemo } from 'react'
+import { isValidStudent } from 'utils'
+
+export const useTIPStudents = (
+	students: RootDBState['students'],
```
Comment (mudssrali) [inline web/src/hooks/useTIPStudents.ts:5]: ❤️

File: web/src/modules/TIP/HTDashboard/Lessons/index.tsx [hunk 3]
```diff
@@ -80,13 +85,7 @@ const LessonDashboard = () => {
...
 			{Object.entries(curriculum ?? {})
 				.filter(([level, _]) => level.startsWith('Level'))
-				.sort(([level_a, _], [level_b, _noUse]) =>
-					level_a == 'Level KG'
-						? -1
-						: level_b == 'Level KG'
-						? 1
-						: level_a.localeCompare(level_b)
-				)
+				.sort(([level_a, _], [level_b, _noUse]) => levelSorter(level_a, level_b))
```
Comment (mudssrali) [inline web/src/modules/TIP/HTDashboard/Lessons/index.tsx:88]: We can simply do as we only need array first element:
```tsx
sort(([levelA], [levelB]) =>...)
```
OR
```tsx
sort(([a], [b]) =>...)
```

File: web/src/modules/TIP/HTDashboard/Lessons/index.tsx [hunk 4]
```diff
@@ -95,10 +94,10 @@ const LessonDashboard = () => {
...
 									'w-full px-2 py-1 text-white font-medium text-lg flex-row flex space-x-2 items-center',
 									{
-										'bg-blue-tip-brand': level === 'Level KG',
-										'bg-yellow-tip-brand': level === 'Level 1',
-										'bg-green-tip-brand': level === 'Level 2',
-										'bg-orange-tip-brand': level === 'Level 3'
+										'bg-blue-tip-brand': level === TIPLevelMap.KG,
+										'bg-yellow-tip-brand': level === TIPLevelMap.L1,
+										'bg-green-tip-brand': level === TIPLevelMap.L2,
+										'bg-orange-tip-brand': level === TIPLevelMap.L3
```
Comment (mudssrali) [inline web/src/modules/TIP/HTDashboard/Lessons/index.tsx:100]: nice.

File: web/src/modules/TIP/HTDashboard/Quizzes/index.tsx [hunk 5]
```diff
@@ -65,13 +70,7 @@ const QuizDashboard = () => {
...
 			{Object.entries(quizzes ?? {})
 				.filter(([level, _]) => level.startsWith('Level'))
-				.sort(([level_a, _], [level_b, _noUse]) =>
-					level_a == 'Level KG'
-						? -1
-						: level_b == 'Level KG'
-						? 1
-						: level_a.localeCompare(level_b)
-				)
+				.sort(([level_a, _], [level_b, _noUse]) => levelSorter(level_a, level_b))
```
Comment (mudssrali) [inline web/src/modules/TIP/HTDashboard/Quizzes/index.tsx:73]: same goes here. However you can create a `comparator-function` as
```tsx
export const compareLevel([a], [b]) => {...}
```

File: web/src/modules/TIP/HTDashboard/Quizzes/index.tsx [hunk 6]
```diff
@@ -80,10 +79,10 @@ const QuizDashboard = () => {
...
 									'w-full px-2 py-1 text-white font-medium text-lg flex-row flex space-x-2 items-center',
 									{
-										'bg-blue-tip-brand': level === 'Level KG',
-										'bg-yellow-tip-brand': level === 'Level 1',
-										'bg-green-tip-brand': level === 'Level 2',
-										'bg-orange-tip-brand': level === 'Level 3'
+										'bg-blue-tip-brand': level === TIPLevelMap.KG,
+										'bg-yellow-tip-brand': level === TIPLevelMap.L1,
+										'bg-green-tip-brand': level === TIPLevelMap.L2,
+										'bg-orange-tip-brand': level === TIPLevelMap.L3
```
Comment (mudssrali) [inline web/src/modules/TIP/HTDashboard/Quizzes/index.tsx:85]: here we go.

File: web/src/modules/TIP/HTDashboard/Tests/ComparisonMatrix/index.tsx [hunk 7]
```diff
@@ -171,15 +158,14 @@ const MatrixGroupView = ({ filters }: GroupViewProps) => {
 				return {
 					...agg,
 					[res.type]: {
-						...calculated_result,
-						percent: Math.round((calculated_result.obt / calculated_result.total) * 100)
+						...calculatedResult,
+						percent: Math.round((calculatedResult.obt / calculatedResult.total) * 100)
```
Comment (mudssrali) [inline web/src/modules/TIP/HTDashboard/Tests/ComparisonMatrix/index.tsx:162]: I feel `calculatedResult.obt` is previously done.

File: web/src/modules/TIP/HTDashboard/Tests/MidpointTest/index.tsx [hunk 8]
```diff
@@ -52,19 +61,16 @@ const HTMidpoint = () => {
 		(state: RootReducerState) => state.targeted_instruction
 	)
 
-	const [view, setView] = React.useState('')
+	const [view, setView] = React.useState(undefined)
```
Comment (mudssrali) [inline web/src/modules/TIP/HTDashboard/Tests/MidpointTest/index.tsx:64]: nice.

File: web/src/utils/TIP.ts [hunk 9]
```diff
@@ -728,3 +728,7 @@ export const getSkillViewQuizResult = (
 		]
 	}, [])
 }
+
+export const levelSorter = (level_a: string, level_b: string) => {
+	return level_a == 'Level KG' ? -1 : level_b == 'Level KG' ? 1 : level_a.localeCompare(level_b)
+}
```
Comment (mudssrali) [inline web/src/utils/TIP.ts:734]:
```tsx
export const compareLevels([a], [b]) => {
   a == TIPLevelMap.KG ? -1 : b == TIPLevelMap.KG ? 1 : a.localeCompare(b)
}
```
and we can directly pass this to `sort()` like
```tsx
sort(compareLevels)
```

**Review by mudssrali — CHANGES_REQUESTED** [review]
Thanks @hmzahmed a lot of improvement, Get ahead! just a change required for level sorting, compare function.

File: web/src/modules/TIP/HTDashboard/Tests/ComparisonMatrix/index.tsx (hunk 7 above)
Comment (hmzahmed) [reply web/src/modules/TIP/HTDashboard/Tests/ComparisonMatrix/index.tsx:163]: I dont think so

**Review by mudssrali — APPROVED** [review]

**Pull request merged**